packages/utils/src/connectWorkers.ts

```diff
@@ -1,22 +1,26 @@
+import { isNodejs } from './checkEnvironemnt';
+
 const workersMap: { [key: string]: Worker } = {};
 
-// @ts-ignore
-if (typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope) {
-  console.warn(`Don't use this on webworkers, only on the main thread`);
-} else {
-  addEventListener('message', (ev) => {
-    if (ev && ev.data && !ev.data.workerId) {
-      if (ev.data.detail) {
-        ev.data.workerId = 1;
-        const propogateTo = workersMap[ev.data.detail.to] || workersMap[ev.data.detail.address]; // discoveryEvents || rsocketEvents
-        if (propogateTo) {
-          // console.log('window -> propogateTo', ev.data);
-          // @ts-ignore
-          propogateTo.postMessage(ev.data, ev.ports || undefined);
+if (!isNodejs()) {
+  // @ts-ignore
+  if (typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope) {
+    console.warn(`Don't use this on webworkers, only on the main thread`);
+  } else {
+    addEventListener('message', (ev) => {
+      if (ev && ev.data && !ev.data.workerId) {
+        if (ev.data.detail) {
+          ev.data.workerId = 1;
+          const propogateTo = workersMap[ev.data.detail.to] || workersMap[ev.data.detail.address]; // discoveryEvents || rsocketEvents
+          if (propogateTo) {
+            // console.log('window -> propogateTo', ev.data);
+            // @ts-ignore
+            propogateTo.postMessage(ev.data, ev.ports || undefined);
```

Comment:
why do you need ev.ports || undefined? If no ports are available, it should be undefined anyway